### ISSUE TITLE
Fix (remove dash) lines beginning with dash (-)

### ISSUE THREAD

**Issue by taxen** [SubtitleEdit/subtitleedit#2850]
Fix (remove dash) lines beginning with dash (-) option in Fix common errors does not completely work in case such as...

100
00:14:16,560 --> 00:14:18,477
This line ends with a dash-

101
00:14:18,760 --> 00:14:22,116
- And this line begins with a dash.

I see why the option does not work. SE takes the dash in line 101 as continuing mark from the line 100.
However, I would like to remove all the dashes at the beginning if they are not a dialog.

Tested on 3.5.6 build 102. build 107 as well.

Regards,

**Comment by niksedk** [comment]
@taxen: Take a look at Edit -> Multiple replace. Here you can create your own "Fix common errors" via regular expressions, like "^- ":
![image](https://user-images.githubusercontent.com/1261331/37833880-cb25dc7c-2eac-11e8-9b94-f8b2432452cb.png)

**Comment by mkver** [comment]
How about using negative lookahead and lookbehinds? Search for (?s)(?<!^-.*)^-\p{Zs}?(?!.*\n-) and replace it with nothing.

**Comment by taxen** [comment]
@mkver thank you! The regular expression does what I want to do exactly!

I want to make two more replaces.

1) Fix all trailing "-" or "," into "..."

100
00:14:16,560 --> 00:14:18,477
This line ends with a dash-

100
00:14:16,560 --> 00:14:18,477
This line ends with a comma,

into

100
00:14:16,560 --> 00:14:18,477
This line ends with three dots...

2) Fix an incomplete dialog into a dialog before and after removing line breaks

100
00:14:16,560 --> 00:14:18,477
This line ends with a dash. - This is second line.

or

100
00:14:16,560 --> 00:14:18,477
This line ends with a dash. <br />- This is second line.

into

100
00:14:16,560 --> 00:14:18,477
\- This line ends with a dash.
\- This is second line.

For 1), I made four regular expressions in multiple replacement.
![re1](https://user-images.githubusercontent.com/3875423/37864040-223546ce-2fac-11e8-9d36-8aeda38d60b9.png)

For 2), I made one regular expression... from "(?<!^-.*)([^-]*) - " to "- $1\n- "
![re2](https://user-images.githubusercontent.com/3875423/37864115-87a6f358-2fad-11e8-90c8-d1d631fe0c75.png)

Are they enough and complete? I am quite new with SE's Regex.
